packages/indexer/src/indexers/autocrat/autocrat-proposal-indexer.ts

```diff
@@ -117,13 +135,33 @@ export const AutocratProposalIndexer: IntervalFetchIndexer = {
         for (const token of [basePass, baseFail, quotePass, quoteFail]) {
           const metadata = await enrichTokenMetadata(token, provider);
           const storedMint = await getMint(provider.connection, token);
+          // NOTE: THIS IS ONLY FOR PROPOSALS AND ONLY FOR BASE / QUOTE CONDITIONAL
+          // TODO: Need to update this for the symbol, name, and image_url base on pass / fail
+          const isQuote = [quoteFail, quotePass].includes(token);
+          const isFail = [quoteFail, baseFail].includes(token);
+          // TODO: Need to get metadata for the base token
+          // TODO: This needs to be refactored
+          let noMetadataName = isFail ? `Proposal ${proposal.account.number}: f${baseTokenMetadata.symbol}` : `Proposal ${proposal.account.number}: p${baseTokenMetadata.symbol}`;
+          let noMetadataSymbol = isFail ? `f${baseTokenMetadata.symbol}` : `p${baseTokenMetadata.symbol}`;
+          let imageUrl;
+          if(isQuote){
+            noMetadataName = isFail ? `Proposal ${proposal.account.number}: fUSDC` : `Proposal ${proposal.account.number}: pUSDC`;
+            noMetadataSymbol = isFail ? `fUSDC` : `pUSDC`;
+          }
 
+          if(dao && daoDetails) {
+            imageUrl = isFail ? daoDetails[0].fail_token_image_url : daoDetails[0].pass_token_image_url;
+            if(isQuote){
+              imageUrl = isFail ? "https://imagedelivery.net/HYEnlujCFMCgj6yA728xIw/6b1ce817-861f-4980-40ca-b55f28f21400/public" : "https://imagedelivery.net/HYEnlujCFMCgj6yA728xIw/f236a0ca-5d7c-4f4a-ca8a-52eb9d72ef00/public";
+            }
```

Comment:
again with the weird branching logic

Reply:
I'm not sure what you mean by this, happy to adjust, just not clear on another way to think about it. There's four strings dependent on if it's a USDC or TOKEN and if it's pass or fail. The above example you had made sense as it was simple string appending / combination.